(maint) Add fips profile

Magisus opened the pull request.

jonathannewman commented:
@Magisus I'm not clear exactly which part is the fix, it looks like security files and call outs for the specific security settings based on jdk for FIPS mode.  Is it the marking of the bc lib as provided?

Magisus commented:
Yeah I think what happened here is that [this commit](https://github.com/puppetlabs/trapperkeeper-metrics/commit/753b19442c160fbd7b4f3e4b9d272b0b0c6a50e1) added BC as a direct dependency (not entirely sure why that was necessary). We hadn't done a release/udpate since that commit went in. That meant that this library would try to pull it in even in FIPS mode, so we were getting some conflicts or bad definitions. I'm not sure _exactly_ what was happening, just that I know from the way all our other libraries are structured these days that pulling in BC that way isn't correct with FIPS in the mix.

I just updated this to remove the explicit dependency on BC, and updated the commit message to describe what I think happened more clearly.

Magisus commented:
I also added an update to travis to run FIPS cells.
